nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java

```diff
@@ -278,6 +289,9 @@ public String getName() {
     }
 
     public void set(String key, Object value) {
+        if (value instanceof Calendar || value instanceof Date) {
```

Comment:
Ah sorry I just realized this is a generic `Document` `set` method and not one on a property we would already know is a date property. On the other hand you could convert from `Calendar` and `Date` to the proper String format expected (to be lenient and helpful to users of the Java client) , unless you plan on doing that in a separate commit.